lib/api/core/models/repositories/tokenRepository.js

```diff
@@ -70,33 +70,43 @@ class TokenRepository extends Repository {
 
   /**
    * @param {User} user
-   * @param {Request} request
-   * @param {object} options
-   * @returns {*}
+   * @param {String} connectionId
+   * @param {Object} options - { algorithm, expiresIn, bypassMaxTTL (false) }
+   *
+   * @returns {Object} { _id, jwt, userId, ttl, expiresAt }
    */
-  generateToken(user, request, options = {}) {
+  generateToken (
+    user,
+    connectionId,
+    {
+      algorithm = this.kuzzle.config.security.jwt.algorithm,
+      expiresIn = this.kuzzle.config.security.jwt.expiresIn,
+      bypassMaxTTL = false
+    } = {}
+  ) {
     if (!user || user._id === null) {
       return securityError.reject('unknown_user');
     }
 
-    if (!request.context.connection.id) {
+    if (!connectionId) {
       return securityError.reject('unknown_connection');
     }
 
-    if (!options.algorithm) {
-      options.algorithm = this.kuzzle.config.security.jwt.algorithm;
-    }
-
-    if (!options.expiresIn) {
-      options.expiresIn = this.kuzzle.config.security.jwt.expiresIn;
+    const parsedExpiresIn = parseTimespan(expiresIn);
+    // @todo test this
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)